### ISSUE TITLE
[Feature request] Consider loosely coupling with netlify template

### ISSUE THREAD

**Issue by vanadium23** [oleeskild/obsidian-digital-garden#41]
Hello, @oleeskild.

I've found your plugin very useful for publishing notes to Github pages. There were many attempts to make fully featured analogue to obsidian publish. For example:
- https://dev.to/joeholmes/creating-a-diy-digital-garden-with-obsidian-and-gatsby-378e
- https://github.com/ppeetteerrs/obsidian-zola
- https://github.com/Maxence-L/notenote.link
- https://github.com/maximevaillancourt/digital-garden-jekyll-template (mostly featured and nearly compatible with the plugin)

Most of them share one common thing: `_notes` folder as main for site. But none of them, done publishing from Obsidian. :+1: 

May be we can customize plugin to support other templates?
Things to make it happen:
- change base notes folder in settings
- feature toggles on note processing (fullLinkName, excalidraw)

**Comment by andrey-jef** [comment]
If I understand correctly, this plugin used 11ty as static site generator, not currently support zola, jekyll or gastby.

And I prefer to deploy via Netlify / Vercel / Render instead of GitHub Pages. Since, iirc, these hosting services allow me to implement server-side redirect, which are not currently supported by GitHub Pages.

**Comment by vanadium23** [comment]
It's not life Netlify vs Github Pages. 
I've done some minor work, and everything is fine with jekyll both on Github Pages:
- https://github.com/vanadium23/digital-garden-jekyll-template
- https://vanadium23.me/digital-garden-jekyll-template/ (Github Pages)
- https://626e7c7fc4048700093e6634--curious-biscotti-5aecac.netlify.app/openbox/home/ (Same but on Netlify)

It's more about working only with template from https://github.com/oleeskild/digitalgarden, but with a little configuration it can work with zola, jekyll and gatsby.

I think the plugin can done only two features:
1. need to render Obsidian markdown with extra features (excalidraw, some default front matter, Dataview (!no other can)).
2. and push to provided repo with specified path.

But it is just an idea. If the goal is to handle all in one basic setup, than current implementation is good.

**Comment by oleeskild** [comment]
This is a great idea. Simply adding a way to specify which folder it should push to should make this compatible with most templates out there. 

I think that it must be a bit "hidden" in the settings so that users that only want something that works out of the box won't change the setting by mistake and break it. Perhaps it could have two modes. One for using it the default way, with the note folder setting hidden, and another for using it in "custom template" mode or something.

Do you have any thoughts on this? How such a setting should be presented to the user?

**Comment by vanadium23** [comment]
I've implemented folder setting: 
https://github.com/vanadium23/obsidian-digital-garden/commit/fe91f93eaf5a0bc5e5d394ce5aacd60734d54a57

But your question leads me to idea with a separate plugin. This can work like periodic notes and calendar plugin.
So core plugin (publisher = render from Obsidian + uploader) supports several providers and plugin, and digital garden plugin is a wrapper with options set by default plus all automation for digitalgarden template.

Or just add tab with advanced options here, but I think it will be too much options:
* feature toggles for all converters
* uploading configs

**Comment by vanadium23** [comment]
Hello, @oleeskild.

Are you okay if I fork the plugin to separate one for idea from the issue? I'll use this repo as upstream, but move it towards published + uploader, rather than all in one solution.

**Comment by oleeskild** [comment]
Absolutely, go ahead. I've had far less time to work on this than I would like, so if you're willing to do something like that, that would be great!

**Comment by oleeskild** [comment]
As there exists another plugin designed to do exactly this, largely based on this codebase, I have decided that this will not be implemented. Anyone looking for this feature can instead use Mara-Li's [Github Publisher](https://github.com/ObsidianPublisher/obsidian-github-publisher).

I'd rather want this project to be as easy as possible to use and setup for those who want an easy-to-use free alternative. Those who want more advanced use-cases can use the plugin linked above.

Thanks for the suggestion though!

**Comment by vanadium23** [comment]
I think your plugin is better, but thanks for the feedback! :+1:

**Comment by oleeskild** [comment]
I thought they were very similar, but from your comment I guess they're not? I haven't actually used the github publisher so perhaps I was a bit quick on closing this one. I'll reopen the issue.
